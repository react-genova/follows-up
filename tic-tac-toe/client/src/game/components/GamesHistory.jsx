import React from 'react';
import PropTypes from 'prop-types';
// import Human from '../../settings/components/Human';
// import Machine from '../../settings/components/Machine';
import { SIGN_X, SIGN_O } from '../../board/modules/types/signs.constants';
import GamesHistorySign from './GamesHistory.sign';
import {
    Results, Row, Name, Score, Table, Title, Wrapper,
} from './GamesHistory.styled';

const GamesHistory = ({
    player1name, player2name, player1score, player2score, player1sign, player2sign, draws,
}) => (
    <Wrapper>
        <Title>GAME HISTORY</Title>
        <Table>
            <Results>
                <Row>
                    <GamesHistorySign type={player1sign} width="5em" />
                    {/* <Human showType={false} thumbSize="2.5em" /> */}
                    <Name>{`${player1name} victories`}</Name>
                    <Score>{player1score}</Score>
                </Row>
                <Row>
                    <GamesHistorySign type={player2sign} width="5em" />
                    {/* <Machine showType={false} thumbSize="2.5em" /> */}
                    <Name>{`${player2name} victories`}</Name>
                    <Score>{player2score}</Score>
                </Row>
                <Row>
                    <GamesHistorySign type={player1sign} />
                    <GamesHistorySign type={player2sign} />
                    <Name>Draws</Name>
                    <Score>{draws}</Score>
                </Row>
            </Results>
            <Row justifyContent="flex-end">
                <Score>{draws + player1score + player2score}</Score>
            </Row>
        </Table>
    </Wrapper>
);

GamesHistory.propTypes = {
    player1name: PropTypes.string,
    player2name: PropTypes.string,
    player1score: PropTypes.number,
    player2score: PropTypes.number,
    player1sign: PropTypes.oneOf([SIGN_X, SIGN_O]),
    player2sign: PropTypes.oneOf([SIGN_X, SIGN_O]),
    draws: PropTypes.number,
};

GamesHistory.defaultProps = {
    player1name: 'Player 1',
    player2name: 'Player 2',
    player1score: 0,
    player2score: 0,
    player1sign: SIGN_X,
    player2sign: SIGN_O,
    draws: 0,
};

export default GamesHistory;
