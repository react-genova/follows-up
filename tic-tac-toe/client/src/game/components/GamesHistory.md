### Empty history

```jsx
<div style={{ backgroundColor: '#333', padding: '1em' }}>
    <GamesHistory />
</div>
```

### Filled history

```jsx
<div style={{ backgroundColor: '#333', padding: '1em' }}>
    <GamesHistory player1score={11} player2score={8} draws={23} />
</div>
```