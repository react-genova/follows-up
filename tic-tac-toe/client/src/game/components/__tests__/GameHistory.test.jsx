import React from 'react';
import { render, cleanup } from 'react-testing-library';
import 'jest-dom/extend-expect';
import GamesHistory from '../GamesHistory';

describe('GamesHistory component', () => {
    afterEach(cleanup);

    it('renders without crashing (empty)', () => {
        const { container } = render(<GamesHistory />);
        expect(container.firstChild).toMatchSnapshot();
    });

    it('renders without crashing (filled)', () => {
        const { container } = render(
            <GamesHistory player1score={11} player2score={8} draws={23} />,
        );
        expect(container.firstChild).toMatchSnapshot();
    });
});
