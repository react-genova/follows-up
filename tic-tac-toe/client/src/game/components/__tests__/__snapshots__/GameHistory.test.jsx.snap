// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GamesHistory component renders without crashing (empty) 1`] = `
<div
  class="sc-gzVnrw hVUXwm"
>
  <span
    class="sc-ifAKCX dWYKsW"
  >
    GAME HISTORY
  </span>
  <div
    class="sc-EHOje grIMnK"
  >
    <div
      class="sc-bZQynM cQrfPv"
    >
      <div
        class="sc-bwzfXH cPrHFb"
      >
        <div
          style="width: 5em; height: 1em;"
        >
          <div
            class="sc-bdVaJa hWaRAZ"
          >
            <svg
              height="100%"
              viewBox="0 0 100 100"
              width="100%"
            >
              <line
                stroke="cyan"
                stroke-width="8"
                x1="0"
                x2="100"
                y1="0"
                y2="100"
              />
              <line
                stroke="cyan"
                stroke-width="8"
                x1="0"
                x2="100"
                y1="100"
                y2="0"
              />
            </svg>
          </div>
        </div>
        <span
          class="sc-htpNat jHttyq"
        >
          Player 1 victories
        </span>
        <span
          class="sc-bxivhb dyMRuN"
        >
          0
        </span>
      </div>
      <div
        class="sc-bwzfXH cPrHFb"
      >
        <div
          style="width: 5em; height: 1em;"
        >
          <div
            class="sc-bdVaJa hWaRAZ"
          >
            <svg
              height="100%"
              viewBox="0 0 100 100"
              width="100%"
            >
              <ellipse
                cx="50"
                cy="50"
                fill="transparent"
                rx="46"
                ry="46"
                stroke="yellow"
                stroke-width="8"
              />
            </svg>
          </div>
        </div>
        <span
          class="sc-htpNat jHttyq"
        >
          Player 2 victories
        </span>
        <span
          class="sc-bxivhb dyMRuN"
        >
          0
        </span>
      </div>
      <div
        class="sc-bwzfXH cPrHFb"
      >
        <div
          style="width: 2.5em; height: 1em;"
        >
          <div
            class="sc-bdVaJa hWaRAZ"
          >
            <svg
              height="100%"
              viewBox="0 0 100 100"
              width="100%"
            >
              <line
                stroke="cyan"
                stroke-width="8"
                x1="0"
                x2="100"
                y1="0"
                y2="100"
              />
              <line
                stroke="cyan"
                stroke-width="8"
                x1="0"
                x2="100"
                y1="100"
                y2="0"
              />
            </svg>
          </div>
        </div>
        <div
          style="width: 2.5em; height: 1em;"
        >
          <div
            class="sc-bdVaJa hWaRAZ"
          >
            <svg
              height="100%"
              viewBox="0 0 100 100"
              width="100%"
            >
              <ellipse
                cx="50"
                cy="50"
                fill="transparent"
                rx="46"
                ry="46"
                stroke="yellow"
                stroke-width="8"
              />
            </svg>
          </div>
        </div>
        <span
          class="sc-htpNat jHttyq"
        >
          Draws
        </span>
        <span
          class="sc-bxivhb dyMRuN"
        >
          0
        </span>
      </div>
    </div>
    <div
      class="sc-bwzfXH cMCOYb"
    >
      <span
        class="sc-bxivhb dyMRuN"
      >
        0
      </span>
    </div>
  </div>
</div>
`;

exports[`GamesHistory component renders without crashing (filled) 1`] = `
<div
  class="sc-gzVnrw hVUXwm"
>
  <span
    class="sc-ifAKCX dWYKsW"
  >
    GAME HISTORY
  </span>
  <div
    class="sc-EHOje grIMnK"
  >
    <div
      class="sc-bZQynM cQrfPv"
    >
      <div
        class="sc-bwzfXH cPrHFb"
      >
        <div
          style="width: 5em; height: 1em;"
        >
          <div
            class="sc-bdVaJa hWaRAZ"
          >
            <svg
              height="100%"
              viewBox="0 0 100 100"
              width="100%"
            >
              <line
                stroke="cyan"
                stroke-width="8"
                x1="0"
                x2="100"
                y1="0"
                y2="100"
              />
              <line
                stroke="cyan"
                stroke-width="8"
                x1="0"
                x2="100"
                y1="100"
                y2="0"
              />
            </svg>
          </div>
        </div>
        <span
          class="sc-htpNat jHttyq"
        >
          Player 1 victories
        </span>
        <span
          class="sc-bxivhb dyMRuN"
        >
          11
        </span>
      </div>
      <div
        class="sc-bwzfXH cPrHFb"
      >
        <div
          style="width: 5em; height: 1em;"
        >
          <div
            class="sc-bdVaJa hWaRAZ"
          >
            <svg
              height="100%"
              viewBox="0 0 100 100"
              width="100%"
            >
              <ellipse
                cx="50"
                cy="50"
                fill="transparent"
                rx="46"
                ry="46"
                stroke="yellow"
                stroke-width="8"
              />
            </svg>
          </div>
        </div>
        <span
          class="sc-htpNat jHttyq"
        >
          Player 2 victories
        </span>
        <span
          class="sc-bxivhb dyMRuN"
        >
          8
        </span>
      </div>
      <div
        class="sc-bwzfXH cPrHFb"
      >
        <div
          style="width: 2.5em; height: 1em;"
        >
          <div
            class="sc-bdVaJa hWaRAZ"
          >
            <svg
              height="100%"
              viewBox="0 0 100 100"
              width="100%"
            >
              <line
                stroke="cyan"
                stroke-width="8"
                x1="0"
                x2="100"
                y1="0"
                y2="100"
              />
              <line
                stroke="cyan"
                stroke-width="8"
                x1="0"
                x2="100"
                y1="100"
                y2="0"
              />
            </svg>
          </div>
        </div>
        <div
          style="width: 2.5em; height: 1em;"
        >
          <div
            class="sc-bdVaJa hWaRAZ"
          >
            <svg
              height="100%"
              viewBox="0 0 100 100"
              width="100%"
            >
              <ellipse
                cx="50"
                cy="50"
                fill="transparent"
                rx="46"
                ry="46"
                stroke="yellow"
                stroke-width="8"
              />
            </svg>
          </div>
        </div>
        <span
          class="sc-htpNat jHttyq"
        >
          Draws
        </span>
        <span
          class="sc-bxivhb dyMRuN"
        >
          23
        </span>
      </div>
    </div>
    <div
      class="sc-bwzfXH cMCOYb"
    >
      <span
        class="sc-bxivhb dyMRuN"
      >
        42
      </span>
    </div>
  </div>
</div>
`;
